Framework/Crystal/src/CentroidPeaks.cpp

```diff
@@ -357,5 +336,23 @@ int CentroidPeaks::findPixelID(std::string bankName, int col, int row) {
   }
 }
 
+void CentroidPeaks::removeEdgePeaks(
+    Mantid::DataObjects::PeaksWorkspace *peakWS) {
+  int Edge = getProperty("EdgePixels");
+  std::vector<int> badPeaks;
+  for (int i = int(peakWS->getNumberPeaks()) - 1; i >= 0; --i) {
```

Comment:
Use `size_t` instead of int like so:
`for (size_t i = peakWS->getNumberPeaks() - 1; i >= 0; --i){`

Also why does `i` decrement instead of increment? Usually for loops increment hence me missing this the first time.

Finally a point of note - never cast using c style casts in C++:
`int foo = int(bar)` is incorrect as if the types are not safely cast-able the compiler will just "fix it" leading to very subtle bugs. Instead use
`int foo = static_cast<int>(bar);` - the compiler will check if this safe to do at compile time and if not (e.g. trying to cast a int to a class for example) emit an error or warning whereas the other will silently compile.

See here for more details: http://cppblogs.blogspot.co.uk/2013/08/c-style-casts.html